diff --git a/rtl/sram_wrapper.sv.tpl b/rtl/sram_wrapper.sv.tpl
index 7f2178c..39df50b 100644
--- a/rtl/sram_wrapper.sv.tpl
+++ b/rtl/sram_wrapper.sv.tpl
@@ -11,7 +11,7 @@
 <%
   base_peripheral_domain = xheep.get_base_peripheral_domain()
   if base_peripheral_domain.contains_peripheral('w25q128jw_controller'):
-    w25 = xheep.get_base_peripheral_domain().get_W25Q128JW_controller()
+    w25 = base_peripheral_domain.get_peripheral('w25q128jw_controller')
     cache = w25.get_cache()
   else:
     cache = 0
@@ -68,7 +68,7 @@ assign pwrgate_ack_no = pwrgate_ni;
         .douta(rdata_o)
     );
   end
-  % endif 
+  % endif
   else begin
     $error("Bank size not generated.");
   end
diff --git a/scripts/generate_sram.tcl.tpl b/scripts/generate_sram.tcl.tpl
index 00be16f..bb21d73 100644
--- a/scripts/generate_sram.tcl.tpl
+++ b/scripts/generate_sram.tcl.tpl
@@ -56,7 +56,7 @@ wait_on_run xilinx_mem_gen_${num_words}_synth_1
 <%
   base_peripheral_domain = xheep.get_base_peripheral_domain()
   if base_peripheral_domain.contains_peripheral('w25q128jw_controller'):
-    w25 = xheep.get_base_peripheral_domain().get_W25Q128JW_controller()
+    w25 = base_peripheral_domain.get_peripheral('w25q128jw_controller')
     cache = w25.get_cache()
   else:
     cache = 0
